app/routes.py

```diff
@@ -1,2 +1,154 @@
-from flask import Blueprint
+import datetime
+from typing import ChainMap
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import request, Blueprint, make_response, jsonify
+from datetime import datetime
+from dotenv import load_dotenv
+import os
+import requests
+
+
+load_dotenv()
+
+
+task_bp = Blueprint("task_bp", __name__, url_prefix="/tasks")
+goal_bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+
+@task_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    task_response = []
+    if request.method == "GET":
+        if request.args.get("sort") == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif request.args.get("sort") == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+        task_response = [task.to_dict() for task in tasks]
+        return jsonify(task_response), 200
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+            return make_response({"details": "Invalid data"}, 400) 
+        new_task = Task(title=request_body["title"],
+                        description=request_body["description"], 
+                        completed_at=request_body["completed_at"])
+        db.session.add(new_task)
+        db.session.commit()
+        return make_response({"task": new_task.to_dict()}, 201)
+
+
+@task_bp.route("/<task_id>", methods=["GET", "DELETE", "PUT"])
+def handle_task(task_id):
+    task = Task.query.get(task_id)
+    if request.method == "GET":
+        if not task:
+            return make_response(f"Book {task_id} not found", 404)
```

Comment:
This is doing similar work to similar lines in each if/elif block (lines 52-53 and 58-59).  I recommend moving those sections out of the if/elif structure and placing it directly under line 46 to DRY the code.